packages/jwt-middleware/test/operation.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const JwtMiddleware = require('../');
+const Assert = require('assert');
+const Shot = require('shot');
+const Lab = require('lab');
+const Hoek = require('hoek');
+const Crypto = require('crypto');
+
+const lab = Lab.script();
+const describe = lab.describe;
+const it = lab.it;
+
+module.exports = { lab };
+
+describe('JWT middleware operation', { parallel: false }, () => {
+
+    it ('shoud skip authn and authz if no wt-authorize-execution is present', done => {
+        const token =  Crypto.randomBytes(32).toString('hex');
+        const run = createMockRunner({
+            meta: {}
+        });
+
+        return run(
+            {
+                headers: {
+                    authorization: `Bearer ${token}`
+                }
+            },
+            error => {
+                Assert.ifError(error);
+
+                return done();
+            }
+        )
+    });
+
+    it ('shoud skip authn and authz if wt-authorize-execution is set to 0', done => {
+        const token =  Crypto.randomBytes(32).toString('hex');
+        const run = createMockRunner({
+            meta: {
+                "wt-authorize-execution": "0"
+            }
+        });
+
+        return run(
+            {
+                headers: {
+                    authorization: `Bearer ${token}`
+                }
+            },
+            error => {
+                Assert.ifError(error);
+
+                return done();
+            }
+        )
+    });
+
+    it ('shoud throw error if wt-authorize-execution is set and no authorization header is present', done => {
+        const token =  Crypto.randomBytes(32).toString('hex');
+        const run = createMockRunner({
+            meta: {
+                "wt-authorize-execution": "1"
+            }
+        });
+
+        return run(
+            {
+                headers: {}
+            },
+            error => {
+                Assert.ok(error);
+                Assert.equal(error.statusCode, 401);
+
+                return done();
+            }
+        )
+    });
+
+    it ('shoud throw error if token is not a real JWT token', done => {
```

Comment:
I'd like to see some testing for situations like:

1. The supplied token is expired.
2. The supplied token has the incorrect issuer.
3. The supplied token has the incorrect audience.
4. The supplied token has scope that is a string superset of the required scope (see @tjanczuk's [comment](https://github.com/auth0/webtask-userspace/pull/5/files#r179813047)). This should fail.
5. The supplied token is missing the required scope.
6. The supplied token has the correct token and is authorized.
7. The supplied token has the correct scopes but fails authentication in some way.

I think you can get test inspiration from many of the review comments. As we accumulate these, we prevent ourselves from later introducing regressions down the line and we give ourselves a nice safety net for refactoring.

Reply:
1-7 are all done and checked in along with few additional ones.